test/index.test.ts

```diff
@@ -0,0 +1,359 @@
+        query: GET_REGISTRATIONS,
+        variables: {
+          labelhash,
+        },
+      })
+      .reply(statusCode, {
+        data: registrationResponse,
+      });
+  }
+}
+
+function nockInfura(
+  method: string,
+  params: any[],
+  response: EthCallResponse | EthChainIdResponse | NetVersionResponse
+) {
+  nock(INFURA_URL.origin)
+    .persist()
+    .post(INFURA_URL.pathname, {
+      method,
+      params,
+      id: /[0-9]/,
+      jsonrpc: '2.0',
+    })
+    .reply(200, response);
+}
+
+/* Test Setup */
+
+const test = avaTest as TestInterface<TestContext>;
+
+test.before(async (t: ExecutionContext<TestContext>) => {
+  nock.disableNetConnect();
+  nock.enableNetConnect(SERVER_URL.host);
+
+  nockInfura('eth_chainId', [], {
+    id: 1,
+    jsonrpc: '2.0',
+    result: '0x04', // rinkeby
+  });
+  nockInfura('net_version', [], {
+    jsonrpc: '2.0',
+    id: 1,
+    result: '4',
+  });
+  nockInfura(
+    'eth_call',
+    [
+      {
+        to: '0x00000000000c2e074ec69a0dfb2997ba6c7d2e1e',
+        data: '0x0178b8bfb9fab6dd33ccdfd1f65ea203855508034652c2e01f585a7b742c3698c0c8d6b1',
+      },
+      'latest',
+    ],
+    {
+      result:
+        '0x0000000000000000000000004d9487c0fa713630a8f3cd8067564a604f0d2989',
+    }
+  );
+  nockInfura(
+    'eth_call',
+    [
+      {
+        to: '0x4d9487c0fa713630a8f3cd8067564a604f0d2989',
+        data: '0x59d1d43cb9fab6dd33ccdfd1f65ea203855508034652c2e01f585a7b742c3698c0c8d6b100000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000015646f6d61696e732e656e732e6e66742e696d6167650000000000000000000000',
+      },
+      'latest',
+    ],
+    {
+      result:
+        '0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000001f68747470733a2f2f692e696d6775722e636f6d2f4a635a45534d702e706e6700',
+    }
+  );
+
+  for (let namehash of Object.values(mockNameHash)) {
+    const { domainResponse, registrationResponse } = mockEntry[namehash];
+    nockGraph(namehash, domainResponse, registrationResponse);
+  }
+
+  t.context.server = http.createServer(app);
+  t.context.prefixUrl = await listen(t.context.server);
+});
+
+test.after.always((t: ExecutionContext<TestContext>) => {
+  t.context.server.close();
+  nock.enableNetConnect();
+});
+
+/* Tests */
+
+test('get welcome message', async (t: ExecutionContext<TestContext>) => {
+  const result = await got('', {
+    prefixUrl: SERVER_URL,
+  }).text();
+  t.deepEqual(result, 'Well done mate!');
+});
+
+test('get /name/:tokenId for domain (wrappertest3.eth)', async (t: ExecutionContext<TestContext>) => {
+  const result = await got(`name/${mockNameHash.wrappertest3}`, {
+    prefixUrl: SERVER_URL,
+  }).json();
+  t.deepEqual(result, mockEntry[mockNameHash.wrappertest3].expect);
+});
+
+test('get /name/:tokenId for subdomain (sub1.wrappertest.eth)', async (t: ExecutionContext<TestContext>) => {
```

Comment:
Maybe better to have the test name more descriptive like 

`get /name/:tokenId for subdomain returns auto generated image`